### PR TITLE
Add form component, install dependencies, and create simple example

### PR DESCRIPTION
**Pull request by kylemh**
# Description of changes
We need to create login/registration flow. To do that, we need to create forms. To do that, we need form components.

- Remove old form components
- Add needed form components
  - Alert (for warnings eventually)
  - Label
  - Input
  - Form
- Add `react-select`, `formik`, and `yup` dependencies

# Important Notes
1. There are many files changes in this PR, but I was very strict about solid commit history on this branch. Please inspect changes on a per-commit basis - your life will be simpler that way.
2. You can see the example form by going to the `/join` route of the staging deployment

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 6471cb0b0a491ac30515f553564ebc4974a2a9eb

https://deploy-preview-359--operation-code-storybook.netlify.com

**Comment by kylemh** [comment]
This will likely damage total-coverage because of all the deleted components, but I will resolve the diff-coverage issue later! Should be easy, as the `RegistrationForm` component has no tests.

File: components/RegistrationForm/RegistrationForm.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    .required('Required')
+    .oneOf([Yup.ref('email')], validationErrorMessages.emailMatch),
+  password: Yup.string()
+    .required('Required')
+    .min(minPasswordCharNum, validationErrorMessages.length(minPasswordCharNum))
+    .test('password-strength', validationErrorMessages.password, isMinPasswordStrength),
+  'confirm-password': Yup.string()
+    .required('Required')
+    .oneOf([Yup.ref('password')], validationErrorMessages.passwordMatch),
+});
```
Comment (nryoung) [inline components/RegistrationForm/RegistrationForm.js]: `Yup` seems really nice, I have never heard of it before.

Comment (kylemh) [reply components/RegistrationForm/RegistrationForm.js]: Yeah we used it at AutoGravity. It's recommended in Formik's docs.

**Pull request merged**